src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -454,7 +454,8 @@ bool TryDeleteRange(out DateTime? next)
                                 RemoveTimeSeriesNameFromMetadata(context, slicer.DocId, slicer.Name);
                             }
 
-                            return true;
+                            changeVector = holder.ChangeVector;
+                            return end < to;
```

Comment:
can we have a failing test for this change?

Reply:
Regarding `return end < to;` :
I don't think the previous causes something to fail because even if it is `return true`, in the next round we'll stop, but I do think it is more clear that way. 
I failed to produce a failing test for `changeVector = holder.ChangeVector;` but in general, since we modified `AppendDeadSegment` to produce a new change vector, we should return it to `IncomingReplicationHandler` in order to merge the change vectors. 
![image](https://user-images.githubusercontent.com/46426884/158576795-7478bf76-e169-49a2-b595-669e676742f6.png)

Reply:
Before we always returned true. now, if end == to we return false.

Reply:
and it's fine, no? because we need to break...